cpp/arcticdb/CMakeLists.txt

```diff
@@ -548,17 +530,25 @@ target_include_directories(arcticdb_ext
         ${LIBMONGOCXX_STATIC_INCLUDE_DIRS}
         ${LIBBSONCXX_STATIC_INCLUDE_DIRS}
         ${BITMAGIC_INCLUDE_DIRS}
-        ${PYTHON_INCLUDE_DIRS}
         ${PYTHON_NUMPY_INCLUDE_DIR}
-        ${AWSSDK_LINK_LIBRARIES}
+        ${AWSSDK_INCLUDE_DIRS}
         )
 
 set_target_properties(arcticdb_ext PROPERTIES
         PREFIX "${PYTHON_MODULE_PREFIX}"
         )
 
+install(TARGETS arcticdb_ext
+        LIBRARY DESTINATION . COMPONENT Python_Lib
+        RUNTIME DESTINATION . COMPONENT Python_Lib)
+
 ## Unit Tests ##
-if(TEST)
+if(${TEST})
+    message("Python for test compilation:")
+    unset(Python_USE_STATIC_LIBS)
+    find_package(Python 3 COMPONENTS Interpreter Development REQUIRED)
```

Comment:
This will require a dedicated academic paper to explain, but the criticla points are:

* the `Development` component can't be found for main code compilation if the only way to satisfy it is a statically linked `libpython.a`
* the `Development` component must be present to link the C++ tests and it can be either static or dynamic at this point
* CMake 3.18+ tries to solve this problem by splitting `Development` into `Module` and `Embed`, but can't assume we have that version